File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-5.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        left, right = 0, 1
        k = 1
        while right < len(nums):
            if nums[left] == nums[right]:
                right += 1
            else:
                nums[k] = nums[right]
                left = right
                right += 1
                k += 1
        return k 

File: Data Structures & Algorithms/max-water-container/submission-5.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let largestArea = 0;
        let left = 0, right = heights.length - 1;
        while (left < right) {
            const deltaX = right - left;
            const deltaY = Math.min(heights[left], heights[right]);
            const area = deltaX * deltaY;
            largestArea = Math.max(largestArea, area);
            if (heights[left] < heights[right]) {
                left++;
            } else {
                right--;
            }
        }
        return largestArea;
    }
}
